### PR TITLE
Added the possibility of serving a model through a gunicorn socket

### PR DESCRIPTION
**Pull request by pedro-munoz**
Added the possibility of serving a model through a gunicorn socket as a cli parameter in mlflow models serve command

## What changes are proposed in this pull request?
 
The possibility of binding gunicorn to a socket instead of a TCP host:port tuple
while serving a model. This allows to serve a model through HTTPS with Nginx 
 
## How is this patch tested?
 
A new cli option is added to mlflow models serve to server the model through a gunicorn socket:
```bash
mlflow models serve -m path_to_model -s socket_name.sock
```

This is thought to be used with Nginx and HTTPS protocol, but it can be easily
tested with the following command (for the Tutorial example):
```bash
curl -X POST -H "Content-Type:application/json; format=pandas-split" --data '{"coluhlorides", "citric acid", "density", "fixed acidity", "free sulfur dioxide", "pH", "residual sugar", "sulphates", "total sulfur dioxide", "volatile acidity"],"data":[[12.8, 0.029, 0.48, 0.98, 6.2, 29, 3.33, 1.2, 0.39, 75, 0.66]]}' --unix-socket /path_to_your_socket/socket_name.sock http://localhost/invocations
```

### Is this a user-facing change? 

- [ ] No. You can skip the rest of this section.
- [X] Yes. Give a description of this change to be included in the release notes for MLflow users.
 
Explanation in the previous description.
 
### What component(s) does this PR affect?
 
- [ ] UI
- [X] CLI 
- [ ] API 
- [ ] REST-API 
- [ ] Examples 
- [ ] Docs
- [ ] Tracking
- [ ] Projects 
- [ ] Artifacts 
- [X] Models 
- [ ] Scoring 
- [ ] Serving
- [ ] R
- [ ] Java
- [ ] Python

### How should the PR be classified in the release notes? Choose one:
 
- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [ ] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [X] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

**Comment by Pedro-M** [comment]
I closed and reopened the PR due to timeout downloading pandas library at your servers (no changes in the code)

**Comment by Pedro-M** [comment]
Lot of problems with timeout... Again failed due to this reason...

File: .github/pull_request_template.md [hunk 1]
```diff
@@ -1,32 +1,42 @@
 ## What changes are proposed in this pull request?
  
-(Please fill in changes proposed in this fix)
```
Comment (smurching) [inline .github/pull_request_template.md]: Let's undo changes to this file - it's maybe a bit confusing, but you can edit your PR description (as you've done) without editing this template file :)

File: mlflow/models/cli.py [hunk 2]
```diff
@@ -27,17 +27,19 @@ def commands():
 @cli_args.MODEL_URI
 @cli_args.PORT
 @cli_args.HOST
+@cli_args.SOCKET
 @cli_args.WORKERS
 @cli_args.NO_CONDA
 @cli_args.INSTALL_MLFLOW
-def serve(model_uri, port, host, workers, no_conda=False, install_mlflow=False):
```
Comment (smurching) [inline mlflow/models/cli.py]: This seems similar to the change discussed in https://github.com/mlflow/mlflow/pull/1428/files#r292708824 - in general we should decide whether it's alright to pass flavor-specific arguments to `mlflow models serve` - I guess we've in some sense already set a precedent for this by adding `--workers` as an argument that Python respects but R ignores. Will talk to some committers & get a decision for you on this.

**Comment by smurching** [comment]
@pedro-munoz sorry for the delay - I think the use case this PR aims to satisfy might already be possible if you set the `GUNICORN_CMD_ARGS` environment variable ([see docs](http://docs.gunicorn.org/en/stable/settings.html))? e.g. if you run `GUNICORN_CMD_ARGS="-b unix:socket_name.sock" mlflow models serve -m path_to_model`? We went with a similar approach to enable custom serving-time configuration of MLflow models packaged as docker containers in https://github.com/mlflow/mlflow/pull/1428

**Comment by pogil** [comment]
Closing as @smurching proposed a solution and we haven't heard back from @pedro-munoz

**Pull request closed without merging**